src/solvers/refinement/string_refinement.cpp

```diff
@@ -795,12 +794,13 @@ static optionalt<exprt> get_array(
     return {};
   }
 
-  unsigned n;
-  if(to_unsigned_integer(to_constant_expr(size_val), n))
+  auto n_opt = numeric_cast<unsigned, exprt>(size_val);
```

Comment:
The counterpart of the above comment: why does this require a second template argument?